app/forms/select_email_form.rb

```diff
@@ -4,28 +4,28 @@ class SelectEmailForm
   include ActiveModel::Model
   include ActionView::Helpers::TranslationHelper
 
-  attr_reader :user, :selected_email_id
+  attr_reader :user, :identity, :selected_email_id
 
   validate :validate_owns_selected_email
 
-  def initialize(user)
+  def initialize(user:, identity: nil)
     @user = user
+    @identity = identity
   end
 
   def submit(params)
     @selected_email_id = params[:selected_email_id]
 
     success = valid?
-    FormResponse.new(success:, errors:)
+    identity.update(email_address_id: selected_email_id) if success && identity
```

Comment:
if we wanted to drop the `&& identity` we could do this
```suggestion
    identity&.update(email_address_id: selected_email_id) if success
```

Reply:
Even though it's more verbose, I think I prefer the explicitness of the current implementation.